packages/cinterop/build.gradle.kts

```diff
@@ -289,6 +301,83 @@ val capiIosArm64 by tasks.registering {
     build_C_API_iOS_Arm64(releaseBuild = isReleaseBuild)
 }
 
+val buildJVMSharedLibs by tasks.registering {
+    buildSharedLibrariesForJVM()
+}
+
+fun Task.buildSharedLibrariesForJVM() {
+    group = "Build"
+    description = "Compile dynamic libraries loaded by the JVM fat jar for supported platforms."
+    val directory = "$buildDir/jvm_fat_jar_libs"
+    doLast {
+        exec {
+            commandLine("mkdir", "-p", directory)
+        }
+        exec {
+            workingDir(project.file(directory))
+            commandLine(
+                "cmake",
+                project.file("src/jvmMain/")
+            )
+        }
+        exec {
+            workingDir(project.file(directory))
+            commandLine("cmake", "--build", ".", "-j8")
+        }
+
+        // copy files (macos)
+        exec {
+            commandLine("mkdir", "-p", project.file("src/jvmMain/resources/jni/macos"))
+        }
+        File("$directory/core/src/realm/object-store/c_api/librealm-ffi.dylib")
+            .copyTo(project.file("src/jvmMain/resources/jni/macos/librealm-ffi.dylib"), overwrite = true)
+        File("$directory/librealmc.dylib")
+            .copyTo(project.file("src/jvmMain/resources/jni/macos/librealmc.dylib"), overwrite = true)
+
+        // TODO add Windows and Linux
+
+        // build hash file
+        genHashFile(platform = "macos", prefix = "lib", suffix = ".dylib")
+    }
+
+    outputs.file(project.file("src/jvmMain/resources/jni/macos/librealmc.dylib"))
+    outputs.file(project.file("src/jvmMain/resources/jni/macos/librealm-ffi.dylib"))
+    outputs.file(project.file("src/jvmMain/resources/jni/macos/dynamic_libraries.properties"))
+}
+
+fun genHashFile(platform: String, prefix: String, suffix: String) {
+    val resourceDir = project.file("src/jvmMain/resources/jni").absolutePath
+    val libFFI: Path = Paths.get(resourceDir, platform, "${prefix}realm-ffi$suffix")
+    val libRealmc: Path = Paths.get(resourceDir, platform, "${prefix}realmc$suffix")
+
+    // the order matters (i.e 'realm-ffi' first then 'realmc')
+    val macosHashes = """
+            realm-ffi ${sha1(libFFI)}
+            realmc ${sha1(libRealmc)}
+
+    """.trimIndent()
+
+    Paths.get(resourceDir, "macos", "dynamic_libraries.properties").also {
+        Files.writeString(it, macosHashes)
+    }
+}
+
+fun sha1(file: Path): String {
+    val digest = MessageDigest.getInstance("SHA-1")
+    Files.newInputStream(file).use {
+        val buf = ByteArray(16384) // 16k
```

Comment:
No particular reason it's just a power of 2. I don't think this can be shared since this is a kts script (with its own restricted classpath), SoLoader is a sourceset inside a multiplatform module ... the best effort could be to store it as a constant in the `buildSrc` ?